iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/ArmParser.java

```diff
@@ -32,9 +32,13 @@ public class ArmParser implements TreeParser<Tree> {
 
   @Override
   public ArmTree parse(String source, @Nullable InputFileContext inputFileContext) {
-    if (inputFileContext != null && inputFileContext.inputFile.filename().endsWith(".bicep")) {
+    if (inputFileContext != null && isBicepFile(inputFileContext)) {
       return bicepParser.parse(source, inputFileContext);
     }
     return jsonParser.parse(source, inputFileContext);
   }
+
+  public static boolean isBicepFile(InputFileContext inputFileContext) {
+    return inputFileContext.inputFile.filename().endsWith(".bicep");
```

Comment:
```suggestion
    return ArmLanguage.KEY.equals(inputFileContext.inputFile.language());
```
This is not related to your changes, but we should test the language of a file and not a suffix. We would need to update the suffix here, in case the suffix for the language can be customized.

Reply:
And I would prefer to have this logic in the ArmSensor and not in the parser since the ArmMetricsVisitor should not have a direct dependency on the parser.

Reply:
Thanks for the suggestion, I wasn't aware that this information is available in file context. However, it feels somewhat counter-intuitive: it's `ArmLanguage` and not `BicepLanguage`, as if we are implying that there could be more formats countable as ARM sources.